web/source/dom/touchAliasElement.ts

```diff
@@ -30,26 +30,83 @@ namespace com.keyman.dom {
   // If the specified HTMLElement is either a TouchAliasElement or one of its children elements,
   // this method will return the root TouchAliasElement.
   export function findTouchAliasTarget(target: HTMLElement): TouchAliasElement {
+    // The scrollable container element for the before & after text spans & the caret.
+    // Not to be confused with the simulated scrollbar.
     let scroller: HTMLElement;
 
+    // Designed to 'breadcrumb' elements to assist in diagnosis of an evasive known issue.
+    let elementLogger = function() {
+      let elementDescriber = function(element: HTMLElement) {
+        if(element) {
+          let childIndex = undefined;
+          let parent = element.parentElement;
+          if(parent) {
+            for(let child = parent.firstChild, i = 0; child; child = child.nextSibling, i++) {
+              if(child == element) {
+                childIndex = i;
+                break;
+              }
+            }
+          }
+
+          // ID:  should never be specified for TouchAliasElement components
+          // Class:  specified for the root TouchAliasElement and for its text spans
+          // index:  useful for identifying elements with no class info
+          return `tag type ${element.tagName}, class '${element.classList}', index under parent: '${childIndex}', id '${element.id}'`;
+        } else {
+          return `null`;
+        }
+      };
+
+      // Very rough breadcrumbs for use with any Sentry logs.
+
+      // We'd expect to see the telltale 'keymanweb-input' appear as the class for one of the
+      // three elements logged here... at least, normally.
+      console.info(`(For diagnosing https://github.com/keymanapp/keyman/issues/4797)`);
+      // We don't use .innerHTML because that would tell us what the user's been typing.
+      // No keylogging!
+      console.info(`Original target: ${elementDescriber(target)}`);
+      if(target.previousSibling && target.previousSibling instanceof HTMLElement) {
+        console.info(`Previous sibling: ${elementDescriber(target.previousSibling)}`);
+      }
+      if(target.nextSibling && target.nextSibling instanceof HTMLElement) {
+        console.info(`Next sibling: ${elementDescriber(target.nextSibling)}`);
+      }
+      console.info(`Target's parent: ${elementDescriber(target.parentElement)}`);
+      if(target.parentElement instanceof HTMLElement) {
+        console.info(`Target's grandparent: ${elementDescriber(target.parentElement.parentElement)}`);
+      }
+      // Triggers the actual Sentry log (on pages that include Sentry logging)
+      throw new Error(`Unexpected touch-start event target.`);
```

Comment:
Web does not currently have _any_ baked-in Sentry access - not even the 'minimal' version.  That function is therefore unaccessible.

Reply:
> This keeps the existing code path, even if it is currently broken, rather than introducing a new exception code path.

They're the same path.  We simply report the error at a more informative location.